src/services.js

```diff
@@ -205,7 +205,7 @@ export class Services {
   }
 
   /**
-   * @param {!Element|!./service/ampdoc-impl.AmpDoc} elementOrAmpDoc
+   * @param {!Element|!ShadowRoot|!./service/ampdoc-impl.AmpDoc} elementOrAmpDoc
```

Comment:
@jridgewell for approval

Reply:
Can shadowRoots have different CIDs? If so, this is broken, since it'll get the shared ampdoc for the page.

Reply:
shadowRoot gets the same CID. We haven't get request to create new CID for shadowRoot, afaik content in the shadowRoot are from the same domain thus should get the same id. 

I added this to fix type check. Shouldn't affect behavior.

Reply:
@jridgewell Does the sound good to you?